v0/destinations/fbpixel/transform.js

```diff
@@ -0,0 +1,431 @@
+  const { Config } = destination;
+  const { pixelId, accessToken } = Config;
+  const {
+    blacklistPiiProperties,
+    categoryToContent,
+    eventCustomProperties,
+    valueFieldIdentifier,
+    whitelistPiiProperties,
+    limitedDataUSage
+  } = Config;
+
+  const endpoint = `https://graph.facebook.com/v9.0/${pixelId}/events?access_token=${accessToken}`;
+
+  const user_data = constructPayload(
+    message,
+    MAPPING_CONFIG[CONFIG_CATEGORIES.USERDATA.name]
+  );
+  let custom_data = {};
+
+  const commonData = constructPayload(
+    message,
+    MAPPING_CONFIG[CONFIG_CATEGORIES.COMMON.name]
+  );
+  if (category.type !== "identify") {
+    custom_data = {
+      ...custom_data,
+      ...flattenJson(constructPayload(message, MAPPING_CONFIG[category.name]))
+    };
+    custom_data = checkPiiProperties(
+      message,
+      custom_data,
+      blacklistPiiProperties,
+      whitelistPiiProperties,
+      category.standard,
+      eventCustomProperties
+    );
+  } else {
+    custom_data = undefined;
+  }
+  if (category.standard) {
+    custom_data.currency = message.properties.currency || "USD";
+    switch (category.type) {
+      case "product list viewed":
+        custom_data = {
+          ...custom_data,
+          ...handleProductListViewed(message, categoryToContent)
+        };
+        commonData.event_name = "ViewContent";
+        break;
+      case "product viewed":
+        custom_data = {
+          ...custom_data,
+          ...handleProduct(message, categoryToContent, valueFieldIdentifier)
+        };
+        commonData.event_name = "ViewContent";
+        break;
+      case "product added":
+        custom_data = {
+          ...custom_data,
+          ...handleProduct(message, categoryToContent, valueFieldIdentifier)
+        };
+        commonData.event_name = "AddToCart";
+        break;
+      case "order completed":
+        custom_data = {
+          ...custom_data,
+          ...handleOrder(message, categoryToContent, valueFieldIdentifier)
+        };
+        commonData.event_name = "Purchase";
+        break;
+      case "products searched":
+        custom_data = {
+          ...custom_data,
+          search_string: message.properties.query
+        };
+        commonData.event_name = "Search";
+        break;
+      case "checkout started":
+        custom_data = {
+          ...custom_data,
+          ...handleOrder(message, categoryToContent, valueFieldIdentifier)
+        };
+
+        commonData.event_name = "InitiateCheckout";
+        break;
+      default:
+        throw Error("This standard event does not exist");
+    }
+  }
+
+  if (category.type === "page") {
+    commonData.event_name = message.name
+      ? `Viewed Page ${message.name}`
+      : "Viewed a Page";
+  }
+  if (category.type === "simple track") {
+    custom_data.value = message.properties
+      ? message.properties.revenue
+      : undefined;
+    delete custom_data.revenue;
```

Comment:
why we need to delete..will this be populated earlier ?
if it's getting populated, let's have a clear distinct path so that it doesn't happen

Reply:
not handled in transform.js but if present in properties that is mapped in json it should not be there but be present in value field.